File: ADS_2_POO/Python Version/documentosFormatacao.py
import re
class DocumentosFormatacao:
    @staticmethod
    def formata_numero(numero, tamanho_com_digitos, tamanho_sem_digitos):
        numero = DocumentosFormatacao.limpa_string(numero, tamanho_com_digitos)
        numero = DocumentosFormatacao.preencher_numero(numero, tamanho_com_digitos, tamanho_sem_digitos)
        return numero

    @staticmethod
    def preencher_numero(numero, tamanho_com_digitos, tamanho_sem_digitos):
        if len(numero) >= tamanho_sem_digitos:
            return numero

        qtd_zeros = tamanho_com_digitos - len(numero)
        return '0' * qtd_zeros + numero

    @staticmethod
    def limpa_string(numero, tamanho_com_digitos):
        numero = re.sub(r'[^0-9]', '', numero).strip()
        if len(numero) <= tamanho_com_digitos:
            return numero
        return 'Numero Invalido'

File: ADS_2_POO/ADS2/DocumentosFormatacao.cs
using System.Text.RegularExpressions;

public static class DocumentosFormatacao
{
    public static string FormataNumero(string numero, int tamanhoComDigitos, int tamanhoSemDigitos)
    {
        numero = LimpaString(numero, tamanhoComDigitos);
        numero = PreencherNumero(numero, tamanhoComDigitos, tamanhoSemDigitos);

        return numero;
    }

    public static string PreencherNumero(string numero, int tamanhoComDigitos, int tamanhoSemDigitos)
    {
        if (numero.Length >= tamanhoSemDigitos)
        {
            return numero;
        }

        int qtdZeros = tamanhoComDigitos - numero.Length;
        
        return new string('0', qtdZeros) + numero; ;
    }

    public static string LimpaString(string numero, int tamanhoComDigitos)
    {
        numero = Regex.Replace(numero, "[^0-9]", "").Trim();
        if (numero.Length <= tamanhoComDigitos)
        {
            return numero;
        }
        return "Numero Invalido";
    }

}



